documentation/src/docs/metrics.md

```diff
@@ -116,204 +134,169 @@ public final class ServiceStarter {
+dl.incCounterValue(globalMetricName(SEGMENT_WRITE_BYTES), dataLength);
+...
+dl.incCounterValue(SEGMENT_WRITE_BYTES, dataLength, segmentTags(streamSegmentName));
+```
+Here SEGMENT_WRITE_BYTES is the base name of the metric. There are two metrics shown here: one is the global counter
+which has no tags associated, and a segment specific counter which has a list of segment tags associated.
+(Here segmentTags is the method to generate tags based on fully qualified segment name) 
+
+Also we see the required steps to register and use OpStatsLogger(Timer):
 
 1. Get a StatsLogger from MetricsProvider: 
 ```
-StatsLogger STATS_LOGGER = MetricsProvider.getStatsLogger();
+StatsLogger STATS_LOGGER = MetricsProvider.getStatsLogger("segmentstore");
 ```
 2. Register all the desired metrics through StatsLogger:
 ```
-static final OpStatsLogger CREATE_STREAM_SEGMENT = STATS_LOGGER.createStats(SEGMENT_CREATE_LATENCY);
+@Getter(AccessLevel.PROTECTED)
+final OpStatsLogger createStreamSegment = STATS_LOGGER.createStats(SEGMENT_CREATE_LATENCY);
 ```
 3. Use these metrics within code at appropriate place where you would like to collect and record the values.
 ```
-Metrics.CREATE_STREAM_SEGMENT.reportSuccessEvent(timer.getElapsed());
-```
-Here CREATE_STREAM_SEGMENT is the name of this metric, and CREATE_STREAM_SEGMENT is the name of our Metrics logger, it will track operations of createSegment, and we will get the time of each createSegment operation happened, how long each operation takes, and other numbers computed based on them.
-
-### 2.1.1 Output example of OpStatsLogger 
-An example output of OpStatsLogger CREATE_SEGMENT reported through CSV reporter:
-```
-$ cat CREATE_STREAM_SEGMENT.csv 
-t,count,max,mean,min,stddev,p50,p75,p95,p98,p99,p999,mean_rate,m1_rate,m5_rate,m15_rate,rate_unit,duration_unit
-1480928806,1,8.973952,8.973952,8.973952,0.000000,8.973952,8.973952,8.973952,8.973952,8.973952,8.973952,0.036761,0.143306,0.187101,0.195605,calls/second,millisecond
+getCreateStreamSegment().reportSuccessEvent(elapsed);
 ```
+Here SEGMENT_CREATE_LATENCY is the name of this metric, and createStreamSegment is the metric object, which tracks
+operations of createSegment. We will get the time of every createSegment operation happened, how long each operation
+takes, and other numbers computed based on them.
 
 
-### 2.2. Example for Dynamic Gauge and OpStatsLogger(Histogram) 
-This is an example from io.pravega.controller.store.stream.AbstractStreamMetadataStore. In this class, we report a Dynamic Gauge which represents the open transactions and  one histogram (CREATE_STREAM).
+### 2.2. Example for Dynamic Gauge 
+This is an example from io.pravega.controller.metrics.StreamMetrics. In this class, we report 
+a dynamic gauge which represents the open transactions of a stream.
 ```java
-public abstract class AbstractStreamMetadataStore implements StreamMetadataStore  {
+public final class StreamMetrics extends AbstractControllerMetrics implements AutoCloseable {
```

Comment:
And here. Really, I don't understand the need for copy-pasting code into this document.

Reply:
It seems the intention of the original author is to help readers read from beginning to end without the need to leave the doc, by adding(duplicating) some reference contents directly into the doc.
Yes this approach is opposite to java-doc style, where reference link is always enforced, and copy-pasting/duplication is prohibited.

@andreipaduroiu, are we sure we want to change the style drastically?

Meanwhile, I'm removing the less relevant lines from the doc so it only contains the "distilled" metrics operations.

Reply:
Accepted suggestion